docs/Allocene/script/games.js

```diff
@@ -0,0 +1,189 @@
+    shooter: '&category=shooter',
+    arpg: '&category=action-rpg',
+    battleRoyale: '&category=battle-royale',
+    strategy: '&category=strategy',
+    mmorpg: '&category=mmorpg',
+    fighting: '&category=fighting',
+    allgames: '',
+    platform: '&platform=pc',
+    online: '&platform=browser',
+    newGames: '&sort-by=release-date',
+    oldGames: '&sort-by=alphabetical',
+};
+
+const checkNew = document.querySelector('[data-type="check-new"]');
+const checkOld = document.querySelector('[data-type="check-old"]');
+const radioPlatform = document.querySelector('[data-type="platform-radio"]');
+const radioOnline = document.querySelector('[data-type="online-radio"]');
+const searchField = document.querySelector('[data-type="search"]');
+const noFound = document.querySelector('[data-type="text-h2"]');
+
+function genreFilter(params) {
+    const selectElement = document.getElementById('genre');
+    const selectedGenreNow = selectElement.value;
+    return params[selectedGenreNow] || params.allgames;
+}
+
+function buildDynamicUrl(params) {
+    const dynamicUrl = genreFilter(params);
+    const platform = document.querySelector('[data-type="platform-radio"]').checked
+        ? params.platform
+        : '';
+    const online = document.querySelector('[data-type="online-radio"]').checked
+        ? params.online
+        : '';
+    const newGames = document.querySelector('[data-type="check-new"]').checked
+        ? params.newGames
+        : '';
+    const oldGames = document.querySelector('[data-type="check-old"]').checked
+        ? params.oldGames
+        : '';
+
+    return `${params.genre}${dynamicUrl}${platform}${online}${newGames}${oldGames}`;
+}
+
+async function applyFiltersAndRenderCards() {
+    const dynamicUrl = buildDynamicUrl(queryParams);
+    await renderCards(ulContainer, dynamicUrl);
+}
+
+const selectElement = document.getElementById('genre');
+selectElement.addEventListener('change', async () => {
+    await applyFiltersAndRenderCards();
+});
+
+genreFilter(queryParams);
+
+function radioFilter(params) {
+    radioPlatform.addEventListener('change', async () => {
+        params.dynamicUrl = radioPlatform.checked ? params.platform : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+
+    radioOnline.addEventListener('change', async () => {
+        params.dynamicUrl = radioOnline.checked ? params.online : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+}
+
+function checkboxFilter(params) {
+    let dynamicUrl = params.allgames;
+
+    checkNew.addEventListener('change', async () => {
+        dynamicUrl = checkNew.checked ? params.newGames : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+
+    checkOld.addEventListener('change', async () => {
+        dynamicUrl = checkOld.checked ? params.oldGames : params.allgames;
+        await applyFiltersAndRenderCards();
+    });
+
+    renderCards(ulContainer, dynamicUrl);
+}
+
+function searchFilter() {
```

Comment:
```suggestion
searchField.addEventListener('input', onSearch)

function onSearch() {all logic}
```

Reply:
same for all filters

Reply:
I did not understand the point of the edit

Reply:
because we have a criteria
> Не пишіть функції в EventListener. Створіть функцію та викликайте її в лістенері

Reply:
Ah, that's it, I saw what the error was, Done

Reply:
Ok, I'm tired.
You resolved this, and I don't like to spend more time to fight with you.